[media-controls] allow for ::before and ::after augmentation of the container

graouts opened the pull request:
#### 9fd7b9016b84101db831a8ae1650be72567ef5eb
<pre>
[media-controls] allow for ::before and ::after augmentation of the container
<a href="https://bugs.webkit.org/show_bug.cgi?id=241972">https://bugs.webkit.org/show_bug.cgi?id=241972</a>

Reviewed by Devin Rousso.

Since content under the container needs to fade, augmenting the container with ::before and ::after
needs to have the fade animations baked in for new media controls subclasses to be able to use
those pseudo-elements.

* Source/WebCore/Modules/modern-media-controls/controls/media-controls.css:

Canonical link: <a href="https://commits.webkit.org/251863@main">https://commits.webkit.org/251863@main</a>
</pre>

dcrousso commented on Source/WebCore/Modules/modern-media-controls/controls/media-controls.css:
```diff
@@ -90,13 +90,17 @@
 
-.media-controls > * {
+.media-controls > *,
+.media-controls::before,
+.media-controls::after {
```
Do we also need to do this for `.media-controls-container > *`?

NIT: Can we `:is(::before, ::after)`?

graouts replied:
Only the controls themselves ought to fade, so I don't think we want siblings to have those styles. I'll try `:is(::before, ::after)`, you're always one step ahead on those new things!

graouts replied:
Turns out you can't use pseudo elements inside pseudo classes :(

graouts committed:
[media-controls] allow for ::before and ::after augmentation of the container
https://bugs.webkit.org/show_bug.cgi?id=241972

Reviewed by Devin Rousso.

Since content under the container needs to fade, augmenting the container with ::before and ::after
needs to have the fade animations baked in for new media controls subclasses to be able to use
those pseudo-elements.

* Source/WebCore/Modules/modern-media-controls/controls/media-controls.css:

Canonical link: https://commits.webkit.org/251863@main
